test/hooks.test.js

```diff
@@ -2327,6 +2337,149 @@ test('onError hook with setErrorHandler', t => {
   t.end()
 })
 
+test('preDecoding hook should run before parsing and be able to modify the payload', t => {
+  t.plan(5)
+  const fastify = Fastify()
+
+  fastify.addHook('preDecoding', function (req, reply, payload, done) {
+    const modified = new stream.Readable()
+    modified.receivedEncodedLength = parseInt(req.headers['content-length'], 10)
```

Comment:
This length should not be updated to `{ hello: 'another world' }`?

Reply:
Nope. According to the specs, the `Content-Length` must match the transferred data, not the uncompressed one. So I keep the header.